s3torchbenchmarking/README.md

```diff
@@ -0,0 +1,187 @@
+  --shard-size TEXT        If supplied, the images are grouped into tar files
+                           of the given size. Size can be supplied as an IEC
+                           or SI prefix. Eg: 16Mib, 4Kb, 1Gib. Note: these are
+                           case-sensitive notations.
+  --s3-bucket TEXT         S3 Bucket name. Note: Ensure the credentials are
+                           made available either through environment variables
+                           or a shared credentials file.  [required]
+  --s3-prefix TEXT         Optional S3 Key prefix where the dataset will be
+                           uploaded. Note: a prefix will be autogenerated. eg:
+                           s3://<BUCKET>/1k_256x256_16Mib_sharded/
+  --region TEXT            Region where the S3 bucket is hosted.  [default:
+                           us-east-1]
+  --help                   Show this message and exit.
+
+```
+
+Here are some sample dataset configurations that we ran our benchmarks against:
+
+- `-n 20k --resolution 496x387`
+- `-n 20k --resolution 496x387 --shard-size {4, 8, 16, 32, 64}MiB`
+
+Example:
+
+```
+$ s3torch-datagen -n 20k \
+   --resolution 496x387 \
+   --shard-size 4MB \
+   --s3-bucket swift-benchmark-dataset \
+   --region eu-west-2
+
+Generating data: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 100/100 [00:00<00:00, 1243.50it/s]
+Uploading to S3: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:00<00:00, 3378.87it/s]
+Dataset uploaded to: s3://swift-benchmark-dataset/20k_496x387_images_4MB_shards/
+Dataset Configuration created at: ./conf/dataset/20k_496x387_images_4MB_shards.yaml
+Configure your experiment by setting the entry:
+    dataset: 20k_496x387_images_4MB_shards
+Alternatively, you can run specify it on the cmd-line when running the benchmark like so:
+    s3torch-benchmark -cd conf  -m -cn <CONFIG-NAME> 'dataset=20k_496x387_images_4MB_shards'
+```
+
+---
+
+Finally, once the dataset and other configuration modules have been defined, you can kick off the benchmark by running:
+
+    $ cd s3torchbenchmarking/
+
+    $ s3torch-benchmark -cd conf -m -cn YOUR-TEST-CONFIGURATION 
+    
+    # Example-1:
+    $ s3torch-benchmark -cd conf -m -cn dataloading 'dataset.prefix_uri=<S3-PREFIX>' 'dataset.region=eu-west-2'
+
+    # Example-2:
+    $ s3torch-benchmark -cd conf -m -cn checkpointing 'dataset.prefix_uri=<S3-PREFIX>' 'dataset.region=eu-west-2'
+
+_Note: For overriding any other benchmark parameters,
+see [Hydra Overrides](https://hydra.cc/docs/advanced/override_grammar/basic/). You can also run `s3torch-benchmark --hydra-help` to learn more._
+
+Experiments will report total training time, number of training samples as well as host-level metrics like CPU
+Utilisation, GPU Utilisation (if available) etc.
+
+## Next Steps
```

Comment:
Not blocking: It would be great to include instructions on how to configure logging, especially the log level.